crates/sncast/src/lib.rs

```diff
@@ -278,15 +279,27 @@ pub async fn wait_for_tx(
     }
     let retries = timeout / u16::from(retry_interval);
     for i in (1..retries).rev() {
-        match provider.get_transaction_receipt(tx_hash).await {
-            Ok(receipt) => {
-                return match receipt.execution_result() {
-                    ExecutionResult::Succeeded => Ok("Transaction accepted"),
-                    ExecutionResult::Reverted { reason } => {
-                        Err(anyhow!("Transaction has been reverted = {reason}"))
+        match provider.get_transaction_status(tx_hash).await {
+            Ok(status) => match status {
+                starknet::core::types::TransactionStatus::Received => {
```

Comment:
I think we should probably combine this with ` Err(StarknetError(TransactionHashNotFound))`, otherwise we loose an info about retries/timeouts + in both cases we wait for tx acceptance, which at the higher level is very similar scenario

Reply:
I left both information, about the reason for retry and number of retries